tidy3d/plugins/near2far/near2far.py

```diff
@@ -1,130 +1,483 @@
+    """Data structure to store surface monitor data with associated surface current densities."""
+
+    monitor: FieldMonitor = pydantic.Field(
+        ...,
+        title="Field monitor",
+        description=
+        "Object of :class:`.FieldMonitor` on which near fields will be sampled and integrated.",
+    )
+
+    normal_dir: Direction = pydantic.Field(
+        '+',
+        title="Normal vector orientation",
+        description=
+        ":class:`.Direction` of the surface monitor's normal vector w.r.t.\
+ the positive x, y or z unit vectors. Must be one of '+' or '-'.",
+    )
+
+    size: Size = pydantic.Field(
+        None,
+        title="Size",
+        description="Size in x, y, and z directions.",
+        units=MICROMETER,
+    )
+
+    @property
+    def axis(self) -> Axis:
+        """Returns the :class:`.Axis` normal to this surface."""
+        # assume that the monitor's axis is in the direction where the monitor is thinnest
+        return np.argmin(self.monitor.size)
+
+    @pydantic.validator("size", always=True)
+    def set_size(cls, val, values):
+        """Sets the .size field using the size of the :class:`.FieldMonitor`."""
+        val = values.get("monitor").size
+        return val
 
-class Near2Far:
+    _plane_validator = assert_plane()
+
+
+class Near2Far(pydantic.BaseModel):
     """Near field to far field transformation tool."""
 
-    def __init__(self, sim_data: SimulationData, mon_name: str, frequency: float):
-        """Constructs near field to far field transformation object from monitor data.
+    class Config:
+        """Sets config for all :class:`Near2Far` objects.
+
+        Configuration Options
+        ---------------------
+        arbitrary_types_allowed : bool = True
+            Allow types like numpy arrays.
+        """
+        
+        arbitrary_types_allowed = True
+
+    sim_data: SimulationData = pydantic.Field(
+        ...,
+        title="Simulation data",
+        description="Container for simulation data containing the near field monitors.",
+    )
+
+    surfaces: List[Near2FarSurface] = pydantic.Field(
+        ...,
+        title="Surface monitor with direction",
+        description="List of each :class:`.Near2FarSurface` to use as source of near field.",
+    )
+
+    frequency: float = pydantic.Field(
+        ...,
+        title="Frequency",
+        description="Frequency to select from each :class:`.Near2FarSurface` for projection.",
+    )
+
+    pts_per_wavelength: int = pydantic.Field(
+        10,
+        title="Points per wavelength",
+        description="Number of points per wavelength in the background medium with which \
+to discretize the surface monitors for the projection. Default: 10.",
+    )
+
+    medium: Medium = pydantic.Field(
+        None,
+        title="Background medium",
+        description="Background medium in which to radiate near fields to far fields. \
+Default: same as the :class:`.Simulation` background medium.",
```

Comment:
instead of "Default:" just say "if `None`, uses the :class:`.Simulation` background medium"